### ISSUE TITLE
story doesnt always udpate

### ISSUE THREAD

**Issue by kolbdog3333** [p0ody/ff2ebook#48]
Hi when I'm using force update to downlaod a story it does not download all chapters for example, I'm trying to get the latest chapter for this fanfic
 https://www.fanfiction.net/s/13964839/42/Jennifer-Craw-and-the-Gardens-of-Whitebridge-26
But it only has 40 chapters and it was updated an hour ago and it still says 40 chapters while the grabber says 42/42

**Comment by StarWolf3000** [comment]
I have force-updated the story and it should have 42 chapters now. Depending on your ebook reader device/app, you might have to delete the file and reimport it (I'm using Aldiko on Android for reading).

**Comment by kolbdog3333** [comment]
Nope still 40 chapters the file size is still the same.

**Comment by kolbdog3333** [comment]
I just opened the epub its still 40 chapters.

**Comment by kolbdog3333** [comment]
Just did same thing again with this story
 https://www.fanfiction.net/s/13990103/16/Baka-to-Power-Pony-Omniversal
it says 16/16 but when opening the epub their are only 15 chapters

**Comment by kolbdog3333** [comment]
Just did another story been 1 hour since last update did same thing again not sure what problem is.
 https://www.fanfiction.net/s/13720575/92/A-Cadmean-Victory-Remastered

**Comment by p0ody** [comment]
Are you really trying the epubs or mobi ?  I remember i had this trouble with mobi that didnt get updated and i didn't get around to fixing that yet.
When i tried with epubs, i did get it with the right numbers of chapters.

#edit: if that was the case, try again.  I Just deleted all the mobi files so they will get regenerated next time they are requested.

**Comment by kolbdog3333** [comment]
Trying with epub i don't use mobi.

**Comment by p0ody** [comment]
So if you download https://ff2ebook.com/download.php?source=ffnet&id=13964839&filetype=epub, you get only 40 chapters ?

Personnaly i'm getting 42.

![image](https://user-images.githubusercontent.com/564032/146051805-eaa64e90-9a7e-430f-8a6a-807db2a10644.png)

![image](https://user-images.githubusercontent.com/564032/146051771-87e35de2-0f2c-4907-8fe8-0690f37860a7.png)

**Comment by StarWolf3000** [comment]
This is what I get:
![Screenshot_20211214-184543](https://user-images.githubusercontent.com/787808/146051979-501a4d81-06f3-4c37-beee-7868289e169a.png)

**Comment by kolbdog3333** [comment]
correct and for a while it wasn't collecting current chapters not sure why

**Comment by kolbdog3333** [comment]
When downloading 
 https://www.fanfiction.net/s/13720575/92/A-Cadmean-Victory-Remastered
gets all chapters in mobi format but ebub only 91.

**Comment by p0ody** [comment]
this is weird.  I'm getting 92.

Are you certain you are not reopening a file that you downloaded previously ?  Or perhaps a cached version in your reader ?

![image](https://user-images.githubusercontent.com/564032/146052932-bae90c41-b22c-4c33-81ea-a0e627eb1a20.png)

**Comment by kolbdog3333** [comment]
And tried that jennifer craw one again still 40 chapters as epub

**Comment by kolbdog3333** [comment]
im positive because am using force update to download the fanfic

**Comment by kolbdog3333** [comment]
had to clear my cache and it seemed to work this time,

**Comment by p0ody** [comment]
I mean, are you really opening the last file saved on your device ?
Make sure to open the one with the latest number at the end if multiple files with the same name.
![image](https://user-images.githubusercontent.com/564032/146053790-4684886e-9c56-4b91-a4b7-7b695fc614a8.png)

**Comment by p0ody** [comment]
> had to clear my cache and it seemed to work this time,

Ok cool, perfect.

**Comment by kolbdog3333** [comment]
One last question how long do you need to wait once the story updates on fanfiction.net before using ff2ebook

**Comment by p0ody** [comment]
It should be able to get the latest version with "force update" anytime once its up on fanfiction.

Without force update, it checks fanfiction.net only once a day.

**Comment by kolbdog3333** [comment]
Like if you get email updates

**Comment by kolbdog3333** [comment]
does force udpate work for the email update or just the fanfic website.

**Comment by p0ody** [comment]
You mean when you put your email there ?

![image](https://user-images.githubusercontent.com/564032/146055546-34d06c37-242e-4c07-aeb6-7ec001229787.png)


It also sends the last version

**Comment by kolbdog3333** [comment]
no meant when the latest chapter gets sent to email before website updates.

**Comment by kolbdog3333** [comment]
example

**Comment by kolbdog3333** [comment]
sorry couldnt attatch pic

**Comment by p0ody** [comment]
well, the page on fanfiction.net must have the new chapters, if its only sent by email, in that case no, it wont be updated on ff2ebook.com

**Comment by kolbdog3333** [comment]
The fanfic updates but your signed up for recieving email updates on fanfiction.net.

**Comment by kolbdog3333** [comment]
ok so have to the full hour ok thanks for the clarificaion.

**Comment by kolbdog3333** [comment]
Tring to download this one fanfic but its not letting me its a huge file size 24mb when exporting on fichub but want to download it using ff2ebook
here is the fanfic
 https://www.fanfiction.net/s/12715870/1601/The-Loud-House-Revamped

**Comment by StarWolf3000** [comment]
Tha one is whopping 1,600 chapters huge, I think that's a bit too much for the site to handle. And you will get a similar filesize here, if the fichub one is already 24 MB.

**Comment by kolbdog3333** [comment]
thanks anyways thought id ask

**Comment by p0ody** [comment]
damn, 1601 chapters.  It may take a while on ff2ebook.  I'll do something for you, i'll run it local on my own PC, so it should be faster (the server that ff2ebook is running of is slow AF).  I'll get back to you

**Comment by kolbdog3333** [comment]
thanks appreciate it. And the author updates every night around 10 or 11 pm and its the longest fanfic ever and still growing.

**Comment by StarWolf3000** [comment]
I wonder if it makes sense to automatically split into a new file after a certain amount of chapters, maybe with a maximum of 200 per file. For a fic this huge, it would take 9 parts to cover the 1,601 chapters. This would reduce the memory consumption during the generation of one file compared to when generated in its whole.

**Comment by kolbdog3333** [comment]
I think after a certain file size like 10mb then it splits not chapters since Wrecking Limits has 1,035 chapters and its only at 4mb

**Comment by StarWolf3000** [comment]
You don't know the file size before splitting, only after. Also the final file size after compression depends on word (and therefore character) count, which is 13 million in the 1,601 chapters fic. If the other one not only has fewer chapters but probably lesser words on average per chapter, of course its gonna end up smaller.

Which is why I would split on chapter count, not file size. That would also make generation faster, because in the case of split fic files, only the last one would need to be updated again, so only a restricted amount of chapters need to be transfered and converted.

**Comment by kolbdog3333** [comment]
Oh ok thanks for explaining that.

**Comment by p0ody** [comment]
I tried a couple times and for some reason chapter #1000 fail everytime.  Im sorry but i currently dont have the time to look into it.

![image](https://user-images.githubusercontent.com/564032/146065951-14db9f19-c1dc-45f9-b864-9f7eb60be8fd.png)
